### ISSUE TITLE
EXCEPTION_ACCESS_VIOLATION at address 0x008b79b8

### ISSUE THREAD

**Issue by SpheroidDev** [FAForever/fa#2670]
`EXCEPTION_ACCESS_VIOLATION (0xc0000005) at address 0x008b79b8
    attempted to read memory at 0x72676f76`

Crash dialog text, log file and replay file attached.
Playing replay does not reproduce crash.
[crash-msg-r8957215-2018-12-11.txt](https://github.com/FAForever/fa/files/2668882/crash-msg-r8957215-2018-12-11.txt)
[game.uid.8957215.log](https://github.com/FAForever/fa/files/2668883/game.uid.8957215.log)
[8957215-Spheroid.zip](https://github.com/FAForever/fa/files/2668886/8957215-Spheroid.zip)

**Comment by FAETHER** [comment]
I will take a look at it soon.

**Comment by FAETHER** [comment]
This is a memory parser. It searches through whole stack until it finds the pointer with needed value. The whole process can go in an infinite loop, but at some point it got a memory location that does not exist.

                cmp     [eax], ecx
                jz      short loc_8B79BC

loc_8B79B3:                             ; CODE XREF: sub_8B79A0+1A↓j
                mov     eax, [eax]
                add     eax, 4
                cmp     [eax], ecx
                jnz     short loc_8B79B3

This is a given loop.
But the function this loop is in, is part of DoBeat() function from where it is called every tick. However this loop is executed only when the game ui has too many units selected to render. Also moving eip out of the loop seem to not cause any crash. It looks like this very well may be the main dispatcher for commands like move order, and the memory address it looks for is some sort of pointer to DA array.  I've noticed that its triggered when units almost arrive the destination after the move order given. But it happens only when there are a lot of units starting from <100. I did tests only with air units so no idea how that works for others. Seems to happen right on the edge when the air unit is trying to land itself on the ground and go to idle state after move order finished. <- a fact!
So far everything points to be a sim code, but how does it make sense that game does not crash in replay? Well, the only explanation I can give is that replay maybe uses less memory and somehow it did not get corrupted, as it did in real game.
